<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QUsbwall</class>
 <widget class="QMainWindow" name="QUsbwall">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>QUsbwall</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>10</y>
      <width>281</width>
      <height>551</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QGroupBox" name="group_keymgmt">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>261</width>
       <height>511</height>
      </rect>
     </property>
     <property name="title">
      <string>Key management</string>
     </property>
     <widget class="QLineEdit" name="line_vendorid">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>30</y>
        <width>131</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="line_productid">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>50</y>
        <width>131</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="line_serial">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>70</y>
        <width>131</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_vendorid">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Vendor Id</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_productid">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Product id</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_serial">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Serial number</string>
      </property>
     </widget>
     <widget class="QScrollArea" name="scroll_keyflags">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>241</width>
        <height>301</height>
       </rect>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>220</width>
         <height>145</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>145</height>
        </size>
       </property>
       <property name="sizeIncrement">
        <size>
         <width>0</width>
         <height>5</height>
        </size>
       </property>
       <widget class="QWidget" name="verticalLayoutWidget">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>214</width>
          <height>148</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QRadioButton" name="radio_keyadd">
           <property name="text">
            <string>USBWALL_KEY_ADD</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radio_keydel">
           <property name="text">
            <string>USBWALL_KEY_DEL</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="check_setuid">
           <property name="text">
            <string>USBWALL_KEY_ACCESS_SETUID</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="check_read">
           <property name="text">
            <string>USBWALL_KEY_ACCESS_READ</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="check_exec">
           <property name="text">
            <string>USBWALL_KEY_ACCESS_EXEC</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="check_write">
           <property name="text">
            <string>USBWALL_KEY_ACCESS_WRITE</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
     <widget class="QPushButton" name="button_savekey_local">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>450</y>
        <width>99</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Save locally</string>
      </property>
     </widget>
     <widget class="QPushButton" name="button_savekey_flash">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>450</y>
        <width>99</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Save and load</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="group_keyfile">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>491</width>
      <height>211</height>
     </rect>
    </property>
    <property name="title">
     <string>Key manager</string>
    </property>
    <widget class="QListView" name="list_keyfile">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>471</width>
       <height>161</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="button_loadkeyfile">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>180</y>
       <width>99</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Load all keys</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_kernel">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>230</y>
      <width>501</width>
      <height>331</height>
     </rect>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="172">
          <red>0</red>
          <green>0</green>
          <blue>192</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="172">
          <red>0</red>
          <green>0</green>
          <blue>192</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="172">
          <red>0</red>
          <green>0</green>
          <blue>192</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="172">
          <red>0</red>
          <green>0</green>
          <blue>192</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QGroupBox" name="group_modulestatus">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>481</width>
       <height>71</height>
      </rect>
     </property>
     <property name="title">
      <string>USBWall status</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="group_module_listkey">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>481</width>
       <height>211</height>
      </rect>
     </property>
     <property name="title">
      <string>List of loaded keys</string>
     </property>
     <widget class="QColumnView" name="module_listkey_field">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>451</width>
        <height>171</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QLabel" name="label_kernelstate">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>310</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Kernel state</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
    <addaction name="actionLoad_key_file"/>
   </widget>
   <widget class="QMenu" name="menuEdition">
    <property name="title">
     <string>Edition</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionAbout_QUsbwall"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdition"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout_QUsbwall">
   <property name="text">
    <string>About QUsbwall</string>
   </property>
  </action>
  <action name="actionLoad_key_file">
   <property name="text">
    <string>Load key file</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>QUsbwall</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
